Added transfromAlignment parameter to container

Andy-Python-Programmer opened the pull request:
## Description
Added transfromAlignment parameter to container

## Related Issues
#29875

## Tests

I added the following tests:

I have added tests for the transformAlignment parameter in tests/container_test.dart

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

Andy-Python-Programmer committed:
Added transfromAlignment parameter to container

Andy-Python-Programmer committed:
Added transfromAlignment parameter to container

Andy-Python-Programmer committed:
Merge branch 'master' into master

Andy-Python-Programmer committed:
Fix null saftey issues

CareF committed:
migrate animated_placeholder_perf (#64478)

godofredoc committed:
Enable mac sdk tests and remove them from cirrus. (#65896)

HansMuller committed:
Updated references to obsolete Material button classes in benchmarks/macrobenchmarks (#65900)

HansMuller committed:
Updated references to obsolete Material button classes in examples (#65904)

YeungKC committed:
Fix FormFieldState value not in sync with the onChanged value from TextFormField. (#65695)

Commit:
[manual roll] Roll Engine from 1ef10b240e28 to f84e7a019663 (12 revisions) (#65949)

HansMuller committed:
Updated references to obsolete Material button classes in microbenchmarks (#65950)

liyuqian committed:
Mark nonflaky tests as such (#65898)

MichaelRFairhurst committed:
Remove unused 'dart:async' imports. (#65568)

justinmc committed:
TextField constrained layout bug (#65966)

Fix a bug in the layout of some textfields with constrained height and visual density.

a14n committed:
migrate some cupertino files to nullsafety (#65880)

goderbauer commented on packages/flutter/lib/src/widgets/container.dart:
```diff
@@ -356,6 +357,9 @@ class Container extends StatelessWidget {
   /// The transformation matrix to apply before painting the container.
   final Matrix4? transform;
 
+  /// The transform alignment
```
nit: end with a `.`

Also: this doc needs a little more details about what this property does.

goderbauer commented on packages/flutter/lib/src/widgets/container.dart:
```diff
@@ -427,7 +431,7 @@ class Container extends StatelessWidget {
 
     if (transform != null)
-      current = Transform(transform: transform!, child: current);
+      current = transformAlignment == null ? Transform(transform: transform!, child: current) : Transform(transform: transform!, child: current, alignment: transformAlignment!);
```
Can this just be:

```suggestion
      current = Transform(transform: transform!, child: current, alignment: transformAlignment);
```

?

goderbauer commented on packages/flutter/test/widgets/container_test.dart:
```diff
@@ -586,6 +586,52 @@ void main() {
     expect(tapped, false);
   });
 
+    testWidgets('Container transformAlignment', (WidgetTester tester) async {
```
nit: indentation

goderbauer commented on packages/flutter/test/widgets/container_test.dart:
```diff
@@ -586,6 +586,52 @@ void main() {
+    );
+
+    expect(didReceiveTap, isFalse);
+    await tester.tapAt(const Offset(110.0, 110.0));
```
Instead of testing this indirectly via hit test, can we just check that the corners of the transformed widget are aat the expected location?

Andy-Python-Programmer replied:
Hi @goderbauer I am not quite sure how I can do that. I have done rest of the changes

PixelToast replied:
`renderObject.localToGlobal(Offset.zero)` should work, other tests do something similar.

Andy-Python-Programmer replied:
Hi idk what happened suddenly a lot of commits popped up?

thecalamiity committed:
Added sample code to AnimatedAlign (#64698)

bkonyi committed:
Reland "Re-enable the Dart Development Service (DDS) (#64671)" (#65873)

This reverts commit 66b01c1f2905f6b4f845f7283abe3e458351fe2e.

* Add DDS ipv6 support

* Use --dds-port for DDS instead of hijacking --host-vmservice-port

Commit:
Find text containing in tests (#65072)

nero-angela committed:
[ReorderableListView] remove extra margin added after picking up the item (#65080)

dnfield committed:
Make parameter optional (#65444)

Commit:
[flutter_tools] automatically update to latest sw on install (#65784)

If a new service worker is installed, automatically update this behind the scenes. Immediately after a page refresh, the new worker and cache will be available, though the main.dart.js and others will be available sooner due to the cache busting URLS

engine-flutter-autoroll committed:
Roll Engine from f84e7a019663 to 933f811d94ab (8 revisions) (#65975)

darrenaustin committed:
TextSelectionTheme support (step 2 of 3) (#65044)

chunhtai committed:
always adds alert label for alert dialog in Android (#65973)

gaaclarke committed:
Added the machine's architecture to macos doctor results. (#65978)

Andy-Python-Programmer committed:
Fix issues in code style

Commit:
Revert "always adds alert label for alert dialog in Android (#65973)" (#66031)

This reverts commit fd0554ba89e6e2904355bc96622554b6a4ca6bf9.

renyou committed:
Revert "TextField constrained layout bug (#65966)" (#66027)

This reverts commit 10a66b19d5a7b7169602af74795c80e139907266.

Commit:
Revert "TextSelectionTheme support (step 2 of 3) (#65044)" (#66051)

This reverts commit cb92ffc7fa2a3731b98bd4dc7ddfedcc1f50ea6e.

bkonyi committed:
Roll packages to fix #66038 (#66052)

PR #65873 enabled DDS for non-web Flutter applications. This change
surfaced a bug where DDS would not send ServiceRegistered events to VM
service clients when they subscribed to the Service stream. This would
cause tools and tests that rely on these events (e.g., hot reload in
VSCode) to fail.

Fixes #66038.

HansMuller committed:
Replaced reference to obsolete FlatButton button class in SnackBar (#65988)

jmagman committed:
Inform user how to fix permissions when the observatory URL isn't found on iOS 14 (#65977)

MeneDev committed:
Fixed typo (subetting -> subsetting) (#62502)

hpoul committed:
Remove FlutterApplication from app templates. (#64742)

jmagman committed:
Check that header exists instead of content (#65961)

Commit:
Deprecate VelocityTracker default constructor and added VelocityTracker.withKind constructor (#66043)

We've gotten feedback that the VelocityTracker change was disruptive, though it did not break any of the flutter framework or customer tests. In order to make the change non-breaking, PointerDeviceKind parameter can be made optional.

Nevertheless, this parameter should be provided so that the existing touch handlers can use more accurate gestures on mouse/stylus inputs, so we can encourage this by deprecating the default constructor and pointing users towards the VelocityTracker.withKind constructor that takes a non-optional parameter

Commit:
[flutter_tools] use flutter tool handler for dwds resources and precache tool pub dependencies (#65814)

If the tool is downloaded from a precompiled snapshot, or if the backing source files in the pub cache are deleted, the dwds debugging functionality will break as the client.js file cannot be located. Instead use the PackageConfig to verify that package location, downloading if it is missing.

Override the dwds middleware to avoid Isolate.resolvePackageUri

Fixes #53644
Fixes #65475

thecalamiity committed:
Added clipBehavior to Overlay, Flow, AnimatedSize and AndroidView (#65910)

jnschulze committed:
update_dart_sdk.ps1: Ensure Start-BitsTransfer always throws an exception in case of errors (#65929)

stuartmorgan-g committed:
Add VERSIONINFO to the Windows template (#66025)

Adds a VERSIONINFO to Runner.rc in the Windows app template, populated
from the project creation metadata.

Currently the version itself is hard-coded, but it is future-proofed to
allow plumbing the actual version through at build time via preprocessor
defines.

stuartmorgan-g committed:
Update Windows system font change handling (#66021)

The engine now has an explicit API for system font changes, rather than
expecting the WM_FONTCHANGE message to be forwarded to the Flutter view.

stuartmorgan-g committed:
Consider the Windows app template stable (#66022)

Removes the template version from the Windows template; the API and
tooling boundary will now be considered stable, so there will no longer
be frequent breaking changes.

Also updates the link for adding desktop support to a project for all
three platforms to reflect the current location.

Fixes https://github.com/flutter/flutter/issues/52748

Commit:
[flutter_tools] update windows config feature (#65867)

mzdm committed:
Minor docs updates (#64966)

xu-baolin committed:
Fix the inconsistency between the local state of the input and the engine state (#65754)

younghwan committed:
Apply darkmode style (#65182)

godofredoc committed:
Enable pre/post submit hostonly tests. (#65985)

mdebbar committed:
[web] Inform the engine when read-only flag is flipped (#65499)

darrenaustin committed:
Reland ##65044 (#66061)

Turn the opt-in default for ThemeData.useTextSelectionTheme to true so that everything uses the new TextSelectionTheme by default.

DanielEdrisian committed:
Updated androidMissingSdkInstructions error message (#65964)

yjbanov committed:
Hide flutter test --platform (#65984)

engine-flutter-autoroll committed:
2abe69c60 Roll Dart SDK from 83365979ea85 to e41a6008746d (1 revision) (flutter/engine#21224) (#65987)

a14n committed:
remove non-nullability on Navigator methods (#65997)

chunhtai committed:
reland always adds alert label for alert dialog in Android (#66057)

Commit:
[flutter_tools] optimize fetch requests and remove main.dart.js bypass (#66069)

The main.dart.js bypass is not needed now that we have skipWaiting. Additionally optimize the fetch handler so that resources not in the cache skip the service worker altogether.

Fixes #66068

gusghrlrl101 committed:
Change LicensePage's loading color from scaffoldBackgroundColor to cardColor (#64639)

* wrap LicensePage's loading widget with cardColor

* Set AnimatedSwitcer's duration in LicensePage to zero

* Remove AnimatedSwitcher

* Add test code for checking color is same

dnfield committed:
Fix test file name (continous -> continuous) (#66081)

bkonyi committed:
Ensure VmService instance is disposed after failed direct connection attempt (#66123)

_attemptServiceConnection in FallbackDiscovery would fail if the root
library URI was not a package URI even if the VM service connection
attempt was successful. This resulted in a VM service connection being
left alive, causing DDS to fail its connection to the VM service.

Updated _attemptServiceConnection to ensure the VM service instance is
disposed after a non-connection related failure and to allow for root
library URIs with a file scheme.

haeseoklee committed:
Add dart-pad example code for CupertinoSliverRefreshControl (#65164)

stuartmorgan-g committed:
Remove the Windows 'flutter create' warning (#66113)

darrenaustin committed:
Add .DS_Store to ignore list for the analyzer to run on macos cleanly. (#65530)

speaking-in-code committed:
Make sure all isolates start during flutter driver tests. (#65703)

yash1200 committed:
Changed title to label in bottom_navigation_bar_test.dart (#65918)

Piinks committed:
Add versioning to gold endpoints (#66054)

xu-baolin committed:
Fix bug when updating the `divisions` and `value` of the slider at the same time (#65998)

orestesgaolin committed:
Fix mistake in the docs of RouteInformationParser (#66023)

justinmc committed:
Reland "TextField constrained layout bug (#65966)" (#66055)

jmagman committed:
Stream logging from attached debugger on iOS (#66092)

zanderso committed:
Revert "Ensure VmService instance is disposed after failed direct connection attempt (#66123)" (#66164)

This reverts commit 409e994cd03749d5a903f01cf4ec5b3ee074a2ee.

Co-authored-by: Jenn Magder <magder@google.com>

Andy-Python-Programmer committed:
Update tests

Andy-Python-Programmer commented:
I messed up the rebase sorrry for that. I will make a new pr.
